tests/Oryx.BuildImage.Tests/VersionInformationTest.cs

```diff
@@ -32,6 +32,19 @@ public void OryxBuildImage_Contains_VersionAndCommit_Information()
         {
             var agentOS = Environment.GetEnvironmentVariable("AGENT_OS");
             var gitCommitID = Environment.GetEnvironmentVariable("BUILD_SOURCEVERSION");
+            var currentBuildDefinition = Environment.GetEnvironmentVariable("BUILD.DEFINITIONNAME");
+
+            // On a pull request, Azure DevOps does not build the exact version of the code that has
+            // been pushed, but rather a version merged with the target branch; When the target branch 
+            // pointer has advanced, it creates different commit-ids and this test fails. This only 
+            // happens in PR-validation build, so this is a check for that. For more details, refer to
+            // https://github.com/microsoft/azure-pipelines-tasks/issues/9801
+
+            if (currentBuildDefinition.ToLower().Contains("oryx-validation-github"))
+            {
+                gitCommitID = Environment.GetEnvironmentVariable("SYSTEM_PULLREQUEST_SOURCECOMMITID");
```

Comment:
could we do something like

```csharp
GetEnvVar("SYSTEM_PULLREQUEST_SOURCECOMMITID") ?? GetEnvVar("BUILD_SOURCEVERSION")
```

and not check the `DEFINITIONNAME` at all?

Reply:
Yes, I agree with @dorfire ...we should rather depend on the merged commit id instead of build definition name